app/modules/StartupModule.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package modules
+
+/*
+ * Copyright (c) 2022 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import com.google.inject.AbstractModule
+import uk.gov.nationalarchives.omega.editorial.EchoServerBootstrap
+
+import scala.annotation.unused
+
+@unused // Found by Play, via application.conf (play.modules.enabled)
+class StartupModule extends AbstractModule {
+
+  override def configure(): Unit =
+    bind(classOf[EchoServerBootstrap]).asEagerSingleton()
```

Comment:
Does this stop the server when the application is stopped? i.e. can you run the application in the sbt console, and have the echo server stop when the application is stopped? Or is it shutdown when the JVM is stopped?

Reply:
When you stop `sbt`, it appears that the echo server is stopped, in that it no longer echoes; but how graciously, I don't know; if you have a look at  `IOApp.main` it does handle the JVM shutdown; but of course, it's ultimately up to the resources (like the [JMS Client](https://github.com/nationalarchives/ctd-omega-editorial-frontend/blob/3c0dbfc899ac2111e18a8ce83d5b2fba97d73f1e/app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala#L45-L450) how graceful it is.

P.S. This is my first exposure to Cats Effect / IO, so there's probably a tidier way of embedding an IO application.